platform/emdrv/nvm3/src/ccp_flash_intf.c

```diff
@@ -0,0 +1,349 @@
+  // 0);
+  /* The erase function does the erase and takes care of any missing
+   * configurations to successfully erase the sector*/
+  // return 0;
+
+  /* Erases the SECTOR   */
+  DEBUGOUT("\r\n Erase Sector \r\n");
+  RSI_QSPI_SpiErase((qspi_reg_t *)QSPI_BASE, &spi_configs_erase, SECTOR_ERASE,
+                    sector_address, 1, 0);
+
+  /* The erase function does the erase and takes care of any missing
+   * configurations to successfully erase the sector*/
+  return 0;
+}
+
+uint32_t RSI_FLASH_Write(uint32_t address, unsigned char *data,
+                         uint32_t length) {
+  spi_config_t spi_configs_program;
+  uint32_t check_sum = 0;
+  GetQspiConfig(&spi_configs_program);
+
+  /* writes the data to required address using qspi */
+  DEBUGOUT("\r\n Write Data to Flash Memory \r\n");
+  RSI_QSPI_SpiWrite((qspi_reg_t *)QSPI_BASE, &spi_configs_program, 0x2, address,
+                    (uint8_t *)&data[0], length,
+                    256, // 256
+                    _1BYTE, 0, 0, 0, 0, 0, 0);
+
+  return 0;
+}
+
+uint32_t RSI_FLASH_Read(uint32_t address, unsigned char *data, uint32_t length,
+                        uint8_t auto_mode) {
+  spi_config_t spi_configs_program;
+  uint32_t check_sum = 0;
+  GetQspiConfig(&spi_configs_program);
+
+  /* IO_READ - Manual Mode */
+  if (0 == auto_mode) {
+    /* IO Read config */
+    /* Reads from the address in manual mode */
+    DEBUGOUT("\r\n Read Data From Flash Memory Using Manual Mode\r\n");
+    RSI_QSPI_ManualRead((qspi_reg_t *)(QSPI_BASE), &spi_configs_program,
+                        address, (uint8_t *)data, 0 /*_32BIT*/, length, 0, 0,
+                        0);
+  } else { /* DMA_READ - Auto mode */
+    /* Read the data by using UDMA */
+#if 0 // Need to Implement
+  DEBUGOUT("\r\n Read Data From Flash Memory Using DMA \r\n");
+  UDMA_Read();
+  /* Wait till dma done */
+  while (!done)
+    ;
+#endif
+  }
+
+  return 0;
+}
+
+uint32_t RSI_FLASH_Erasechip() {
+  volatile int sector_off_set_addr = 0;
+
+  spi_config_t spi_configs_erase_chip;
+  GetQspiConfig(&spi_configs_erase_chip);
+
+  /*Note : This function will not erase the MBR content of the flash but erases
+   * the rest of the flash from 72K*/
+  //! Initial offset of the flash to be erased.
+  sector_off_set_addr = INIT_FLASH_OFFSET;
+
+  do {
+    ROMAPI_QSPI_API->qspi_spi_erase((qspi_reg_t *)(M4_QSPI_BASE_ADDRESS),
+                                    &spi_configs_erase_chip, SECTOR_ERASE,
+                                    sector_off_set_addr, 1, 0);
+    //! Increment the sector address by 4K
+    sector_off_set_addr = (sector_off_set_addr + (4 * SIZE_1K));
+    //! This loop will be terminated once it reaches to the 1Mb range of memory
+    //! Compare with 4Mb flash size
+  } while (sector_off_set_addr < FLASH_SECTOR_OFFSET);
+  // The erase function does the erase and takes care of any missing
+  // configurations to successfully erase the sector of the flash
+  return 0;
```

Comment:
There isn't any other return condition or value. Maybe the function doesn't need any return.